src/components/GameTile.vue

```diff
@@ -0,0 +1,350 @@
+
+const props = defineProps({
+  tile: {
+    type: Object,
+    required: true,
+    default: () => null,
+  },
+  currency: {
+    type: String,
+    default: "USD",
+    required: true,
+  },
+});
+
+const platformIconStyle = computed(() => {
+  if (props.tile?.platform) {
+    try {
+      const iconUrl = require(`@/assets/GameTile/${props.tile.platform}-logo.png`);
+      return { "--platform-icon": `url(${iconUrl})` };
+    } catch (e) {
+      console.error(`Cannot load icon for platform: ${props.tile.platform}`);
+      return {};
+    }
+  }
+  return {};
+});
+
+function truncateDescription(text, maxLength) {
+  if (text.length <= maxLength) return text;
+  let truncated = text.slice(0, maxLength);
+  const lastSpace = truncated.lastIndexOf(" ");
+  if (lastSpace > 0) {
+    truncated = truncated.slice(0, lastSpace);
+  }
+  return truncated + "...";
+}
+
+const addToCartButton = reactive({
+  iconSource: require(`@/assets/GameTile/shopping-cart_white.png`),
+  iconAlternativeText: "shopping-cart_white.png",
+  label: "ADD TO CART",
+  disabled: false,
+});
+
+function isDiscount() {
+  const discountInPercentage = props.tile?.discountInPercentage ?? 0;
+  return discountInPercentage > 0;
+}
+
+function discountedPrice() {
+  let discountInPercentage = Math.max(
+    Math.min(props.tile?.discountInPercentage, 100),
+    1
+  );
+  return Math.max(
+    props.tile?.price - (props.tile?.price * discountInPercentage) / 100,
+    0.01
+  ).toFixed(2);
+}
+
+function handleAddToCartButtonDisabled() {
+  addToCartButton.disabled = true;
+  addToCartButton.iconSource = require(`@/assets/GameTile/shopping-cart_after.png`);
+  addToCartButton.iconAlternativeText = "shopping-cart_after.png";
+  addToCartButton.label = "ADDED";
+}
+
+function handleNavigateToGamePage() {
+  console.log("Navigate to game page");
+}
+</script>
+
+<template>
+  <div
+    :style="platformIconStyle"
+    :class="$style['game-tile']"
+    @click="handleNavigateToGamePage"
+  >
+    <div :class="$style['game-tile__image-container']">
+      <img
+        :class="$style['game-tile__image']"
+        :src="require(`@/assets/${tile.imagePath}`)"
+        :alt="`Cover art for ${tile.title}`"
+      />
+    </div>
+    <div :class="[$style.content, $style['game-tile__content']]">
+      <div :class="$style['content__title']">{{ tile.title }}</div>
+      <div :class="$style['content__genres']">
+        <template v-for="genre in tile.genres" :key="genre.name">
+          <div :class="$style['genres__genre']">
+            {{ genre.name }}
+          </div>
+        </template>
+      </div>
+
+      <div :class="$style['content__description']">
+        {{ truncateDescription(props.tile?.description, 250) }}
+        <a href="#" :class="$style['content__link']">Read more</a>
+      </div>
+    </div>
+    <div :class="[$style.cart, $style['game-tile__cart']]">
+      <div v-if="isDiscount() === true" :class="$style['cart__price']">
```

Comment:
First of all...
```suggestion
      <div v-if="isDiscount()" :class="$style['cart__price']">
```

Second of all... we use computed value now.
```suggestion
      <div v-if="hasDiscount" :class="$style['cart__price']">
```